### ISSUE TITLE
VS 2022 MAUI template is missing

### ISSUE THREAD

**Issue by Midnayt** [dotnet/maui#1473]
There is no MAUI template in VS 2022(Version 17.0.0 Preview 1.1)

**Comment by Px2uRo** [comment]
but you can also type "dotnet new maui" command ...

**Comment by GeraudFabien** [comment]
Before dotnet new maui you install maui.
You need to have VS installed before with some workpackage (https://docs.microsoft.com/en-us/dotnet/maui/get-started/installation).

``` powershell
dotnet tool install -g Redth.Net.Maui.Check
maui-check
```

**Comment by Midnayt** [comment]
> 
> 
> Before dotnet new maui you install maui.
> You need to have VS installed before with some workpackage (https://docs.microsoft.com/en-us/dotnet/maui/get-started/installation).
> 
> ```powershell
> dotnet tool install -g Redth.Net.Maui.Check
> maui-check
> ```

Repeated all steps, and MAUI didn't appear in VS 2022

**Comment by GeraudFabien** [comment]
Yes that was for using the cli.
For VS I don't know why you doesn't have it. I don't have VS2022

**Comment by Redth** [comment]
Fixed in VS2022 in an upcoming preview, thanks!

**Comment by DuaRobin** [comment]
Please advise on when next preview or VS2022 update be available for Maui Templates?

**Comment by imerzan** [comment]
Used above steps, not working in VS 2022 Preview

**Comment by Midnayt** [comment]
> 
> 
> Fixed in VS2022 in an upcoming preview, thanks!

New release has been delivered, but MAUI template still is missing!

**Comment by DuaRobin** [comment]
MAUI templates are missing in VS 2022, Preview 2.0 as well.

**Comment by kraknellok** [comment]
Still don't have template in 2022 preview, but do in 2019 preview.

**Comment by Redth** [comment]
Yes, this should be fixed in Preview 3.0.

**Comment by sinsedrix** [comment]
Still no template in VS2019 preview 3.0 and VS2022 preview 2.1

**Comment by Redth** [comment]
Thanks, they aren't expected to appear yet.   VS2019 will not receive any future MAUI previews, and VS2022 may see the templates show up in 3.0 but possibly later.  This is outside of our team's control.

**Comment by Benny56** [comment]
Still not in VS2022 preview 3.0.
`dotnet workload install maui` saved the day. Thanks.

**Comment by moustafashaban** [comment]
I am on Version 17.0.0 Preview 3.1
maui-check looks fine all green
and still no Maui template

**Comment by gplumb** [comment]
This is a bug - but fortunately there is a workaround.  I installed [this](https://marketplace.visualstudio.com/items?itemName=egvijayanand.maui-templates) extension and it not only showed the MAUI extension types it creates (that mirrors their command line equivalents), but _also_ triggered VS to display the Microsoft ones that maui-check is creating.

If I was a betting man, I'd wager that a component or internal registry entry for the "MAUI" project type is incorrect or missing...

**Comment by moustafashaban** [comment]
I solved it by uninstalling the [extension ](https://marketplace.visualstudio.com/items?itemName=ProjectReunion.MicrosoftSingleProjectMSIXPackagingToolsDev17) and reinstall it again

**Comment by mpnow** [comment]
> I solved it by uninstalling the [extension ](https://marketplace.visualstudio.com/items?itemName=ProjectReunion.MicrosoftSingleProjectMSIXPackagingToolsDev17) and reinstall it again

@moustafashaban's fix worked for me as well.

**Comment by GihanSenanayake** [comment]
Thank you very much everyone for the information provided, once I installed the mentioned extension MAUI template started showing up!

**Comment by korhun** [comment]
I got it working with the following steps for VS 2022:
- First I've tried [these steps](https://docs.microsoft.com/en-us/dotnet/maui/get-started/first-app?pivots=windows) but there was no project template for MAUI projects. 
- Then I've tried to uninstall and install [this extension](https://marketplace.visualstudio.com/items?itemName=ProjectReunion.MicrosoftSingleProjectMSIXPackagingToolsDev17). Still not working. 
- Then I've installed [this](https://marketplace.visualstudio.com/items?itemName=egvijayanand.maui-templates) and it worked.

**Comment by Shujee** [comment]
VS2022 final release also doesn't have MAUI templates. I was able to get the templates by running `dotnet new --install Microsoft.Maui.Templates` command in PowerShell window, but creating a new default MAUI project didn't compile and showed 20+ errors, mostly about types like `Grid`, `ScrollViewer`, `ContentPage` etc. being undefined. I then went ahead and installed the full MAUI workload (`dotnet workload install maui`). Took ages and finished successfully, but I still see the same issue in VS2022, i.e. MAUI elements being undefined.
